app/shops.py

```
@@ -0,0 +1,26 @@
+from app.product_cart import ProductCard
+from dataclasses import dataclass
+from app.check_if_exist import check_values
+
+
+@dataclass
+class Shops(ProductCard):
+    name: str
+    location: list
+
+    @classmethod
+    def create_list_of_shops(cls, dataset: dict) -> "Shops":
```

This method has the same two issues as its counterpart in the `Customers` class:

1.  The name `create_list_of_shops` is misleading as it creates a single `Shops` instance, not a list. `from_dict` would be more accurate.

2.  More importantly, if the validation checks on line 21 fail, this method implicitly returns `None`. This will be appended to the list of shops in `create_variales.py`, leading to an `AttributeError` later. The method should raise an exception, such as `ValueError`, when the configuration data for a shop is invalid.